src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/AwsCli.kt

```diff
@@ -4,20 +4,28 @@ import com.atlassian.performance.tools.aws.api.StorageLocation
 import com.atlassian.performance.tools.infrastructure.api.os.Ubuntu
 import com.atlassian.performance.tools.ssh.api.SshConnection
 import java.time.Duration
+import java.util.concurrent.ConcurrentHashMap
 
 internal class AwsCli {
+    private companion object {
+        private val LOCKS = ConcurrentHashMap<String, Any>()
+    }
+
     fun ensureAwsCli(ssh: SshConnection) {
-        if (!ssh.safeExecute("aws --version").isSuccessful()) {
```

Comment:
If we move the old code into a private method, then we win one indentation and a cleaner diff